src/main/java/com/google/devtools/build/lib/remote/common/RemoteCacheClient.java

```diff
@@ -143,6 +143,18 @@ ListenableFuture<Void> downloadBlob(
    */
   ListenableFuture<Void> uploadFile(RemoteActionExecutionContext context, Digest digest, Path file);
 
```

Comment:
Suggestion: instead of requiring every client to provide two overloads of each method, add default methods for the "no force argument" overload here, so that clients only need to implement the other overload.

(It's also not super clear to me why we need `force = true` at all; the only usage I could find outside of tests is in RemoteRepositoryRemoteExecutor.java, and code history doesn't offer an explanation. But let's not deal with it in this PR - I'll investigate it separately.)

Reply:
I didn't want to deal with the change of making RemoteCacheClient an abstract class. I feel that can be a follow up change.